src/Explorer/index.jsx

```diff
@@ -448,17 +466,94 @@ class Explorer extends React.Component {
+    const queryableFields = this.state.queryableFieldsForEachSubcommons[subcommonsURL];
+    Object.keys(filtersApplied).forEach((field) => {
+      if (!queryableFields.includes(field)) {
+        delete filtersAppliedReduced[field];
+      }
+    });
+    return filtersAppliedReduced;
+  }
+
+  downloadSubcommonsData(subcommonsConfig, filtersApplied) {
+    const subcommonsURL = subcommonsConfig.URL;
+    const subcommonsName = subcommonsConfig.name;
+    const datasetIsSelected = this.isDatasetSelected(filtersApplied, subcommonsName);
+    if (!datasetIsSelected) {
+      return [];
+    }
+    const filtersAppliedReduced = this.validateFilterForSubCommons(
+      filtersApplied, subcommonsConfig);
+
+    // FIXME: type is "subject", and fields are all from fieldMapping config.
+    // Need a future refactoring task to remove lots of hardcodings from this file.
+    const type = 'subject';
+    const fields = config.dataExplorerConfig.fieldMapping
+      .map(e => e.field)
+      .filter(f => this.state.queryableFieldsForEachSubcommons[subcommonsURL].includes(f));
+    const queryBody = { type, fields };
+    if (filtersAppliedReduced) queryBody.filter = getGQLFilter(filtersAppliedReduced);
+    return fetchWithCredsAndTimeout({
+      path: `${subcommonsURL}${flatModelDownloadRelativePath}`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(queryBody),
+    }, 3000).then(response => (response.data ? response.data
+      .map(d => ({ ...d, dataset: subcommonsName })) : []));
+  }
+
+  downloadData = () => {
+    this.setState({ isDownloadingData: true });
+    const promiseArray = [];
+    const n = Object.keys(config.subcommons).length;
+    for (let j = 0; j < n; j += 1) {
+      promiseArray.push(
+        this.downloadSubcommonsData(config.subcommons[j], this.state.filter),
+      );
+    }
+    const thisCommonsName = 'ImmPort';
+    promiseArray.push(
+      this.downloadSubcommonsData({
+        URL: '/',
+        name: thisCommonsName,
+      }, this.state.filter),
+    );
+    Promise.all(promiseArray).then((res) => {
+      const allData = res.reduce((acc, cur) => acc.concat(cur), []);
+      const blob = new Blob([JSON.stringify(allData, null, 2)], { type: 'text/json' });
+      const filename = 'clinical.json';
+      FileSaver.saveAs(blob, filename);
+      this.setState({ isDownloadingData: false });
+    });
   }
 
   render() {
     return (
       <React.Fragment>
-        <div className='ndef-page-title'>
-          Data Explorer
-        </div>
         <div id='def-spinner' className={this.state.loading ? 'visible' : 'hidden'} >
           <Spinner />
         </div>
+        {
+          this.state.isUserLoggedIn || (
+            <div className='explorer-visualization__login'>
+              <Button
+                className='explorer-visualization__login-btn'
+                label='see more data'
```

Comment:
```suggestion
                label='See more data'
```
or maybe "Login to see more data" or just "Login"